include/albatross/src/models/ransac_gp.hpp

```diff
@@ -36,11 +36,30 @@ inline
     using GPFitType = typename FitAndIndices<ModelType, FeatureType>::FitType;
     const GPFitType fit(train_dataset.features, train_cov,
                         train_dataset.targets);
-    FitAndIndices<ModelType, FeatureType> fit_and_indices = {fit, inds};
+    const FitAndIndices<ModelType, FeatureType> fit_and_indices = {fit, inds};
     return fit_and_indices;
   };
 }
 
+template <typename ModelType, typename FeatureType>
+inline typename RansacFunctions<
+    FitAndIndices<ModelType, FeatureType>>::IsValidCandidate
+get_gp_ransac_is_valid_candidate(const RegressionDataset<FeatureType> &dataset,
+                                 const FoldIndexer &indexer,
+                                 const Eigen::MatrixXd &cov) {
+  return [&, indexer, cov, dataset](const std::vector<FoldName> &groups) {
+    auto inds = indices_from_names(indexer, groups);
+    const auto train_dataset = subset(dataset, inds);
+    const auto train_cov = symmetric_subset(cov, inds);
+
+    const JointDistribution prior(Eigen::VectorXd::Zero(train_cov.rows()),
+                                  train_cov);
+    const double chi2 = chi_squared_cdf(prior, train_dataset.targets);
+    const double skip_every_1000_candidates = 0.999;
+    return (chi2 < skip_every_1000_candidates);
```

Comment:
The var name is confusing to me. Does it skip every 1000 candidates or skip one on every 1000 candidates? (the later). But it only skips 1/1000 candidates if the candidates are sampled from the prior, so this is really something designed to remove things that don't look like the prior at all. `inclusion_ratio` or `probabilty_threshold` seem like good terms here.
`probability_targets_not_from_prior` or similar could replace `chi2`.
There's also an asymmetry with low chi2 values not being removed.

Reply:
I changed it to `skip_every_1000th_candidate`, that make it more clear?  and I'll change `chi2` to `probability_prior_exceeded`.  I think the asymmetry is OK.  For example, we may want to use a somewhat relaxed prior to perform a cursory check for outliers before proceeding with RANSAC, so I feel like we only want o discard large values.

Reply:
I can understand the code right now, but I still think `skip_every_1000th_candididate`, while more clear, isn't what it's actually doing. It skips every 1000th candidate sampled from the prior, but it removes things that don't look like the prior outright.